spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -0,0 +1,338 @@
+			);
+
+	public static final String JSON_TYPES = "spring_json_header_types";
+
+	private static final List<SimplePatternBasedHeaderMatcher> NEVER_MAPPED = Arrays.asList(
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.ACKNOWLEDGMENT),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.CONSUMER),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.MESSAGE_KEY),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.OFFSET),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.PARTITION_ID),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RAW_DATA),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_MESSAGE_KEY),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_PARTITION_ID),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_TIMESTAMP),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.RECEIVED_TOPIC),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TIMESTAMP),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TIMESTAMP_TYPE),
+		new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TOPIC));
+
+	private final ObjectMapper objectMapper;
+
+	private final List<SimplePatternBasedHeaderMatcher> matchers = new ArrayList<>(NEVER_MAPPED);
+
+	private final Set<String> trustedPackages = new LinkedHashSet<>(DEFAULT_TRUSTED_PACKAGES);
+
+	/**
+	 * Construct an instance with the default object mapper and header patterns for
+	 * outbound headers; all inbound headers are mapped.
+	 * @see #DefaultKafkaHeaderMapper(ObjectMapper)
+	 */
+	public DefaultKafkaHeaderMapper() {
+		this(new ObjectMapper());
+	}
+
+	/**
+	 * Construct an instance with the provided object mapper and default header patterns
+	 * for outbound headers; all inbound headers are mapped. The patterns are applied in
+	 * order, stopping on the first match (positive or negative). Patterns are negated by
+	 * preceding them with "!". The default pattern list is
+	 * {@code "!id", "!timestamp" and "*"}.
+	 * @param objectMapper the object mapper.
+	 * @see org.springframework.util.PatternMatchUtils#simpleMatch(String, String)
+	 */
+	public DefaultKafkaHeaderMapper(ObjectMapper objectMapper) {
+		this(objectMapper, Arrays.asList(
+				"!" + MessageHeaders.ID,
+				"!" + MessageHeaders.TIMESTAMP,
+				"*"));
+	}
+
+	/**
+	 * Construct an instance with the provided object mapper and default header patterns
+	 * for outbound headers; all inbound headers are mapped. The patterns are applied in
+	 * order, stopping on the first match (positive or negative). Patterns are negated by
+	 * preceding them with "!". The patterns will replace the default patterns; you generally
+	 * should not map the {@code "id" and "timestamp"} headers.
+	 * @param objectMapper the object mapper.
+	 * @param patterns the patterns.
+	 * @see org.springframework.util.PatternMatchUtils#simpleMatch(String, String)
+	 */
+	public DefaultKafkaHeaderMapper(ObjectMapper objectMapper, List<String> patterns) {
```

Comment:
For example: `new SimplePatternBasedHeaderMatcher("!" + KafkaHeaders.TOPIC)`
And we have here: ` this.matchers.add(new SimplePatternBasedHeaderMatcher(pattern))`.
So, looks like there is no way to override defaults and, therefore, if we strictly need to set those header to the Kafka message, we should remap them to the custom headers.